File: ai-plane/app/analysis/models/analysis_ingest_event.py
"""Analysis 入库事件模型（与 Java AnalysisIngestEvent camelCase 对齐）。"""

from __future__ import annotations

from datetime import datetime
from typing import Literal
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field


class AnalysisIngestEvent(BaseModel):
    model_config = ConfigDict(populate_by_name=True)

    event_id: UUID | None = Field(default=None, alias="eventId")
    event_type: Literal["ANALYSIS_INGEST"] = Field(default="ANALYSIS_INGEST", alias="eventType")
    job_id: UUID = Field(alias="jobId")
    user_id: int = Field(alias="userId")
    team_id: int | None = Field(default=None, alias="teamId")
    object_key: str = Field(alias="objectKey")
    file_type: str = Field(alias="fileType")
    trace_id: str | None = Field(default=None, alias="traceId")
    created_at: datetime | None = Field(default=None, alias="createdAt")

File: control-plane/src/main/java/com/devops/copilot/modules/file/kafka/event/AnalysisIngestEvent.java
package com.devops.copilot.modules.file.kafka.event;

import java.time.OffsetDateTime;
import java.util.UUID;

/**
 * Kafka analysis.ingest.v1 事件体。
 */
public class AnalysisIngestEvent {

    private UUID eventId;
    private String eventType = "ANALYSIS_INGEST";
    private UUID jobId;
    private Long userId;
    private Long teamId;
    private String objectKey;
    private String fileType;
    private String traceId;
    private OffsetDateTime createdAt;

    public UUID getEventId() { return eventId; }
    public void setEventId(UUID eventId) { this.eventId = eventId; }
    public String getEventType() { return eventType; }
    public void setEventType(String eventType) { this.eventType = eventType; }
    public UUID getJobId() { return jobId; }
    public void setJobId(UUID jobId) { this.jobId = jobId; }
    public Long getUserId() { return userId; }
    public void setUserId(Long userId) { this.userId = userId; }
    public Long getTeamId() { return teamId; }
    public void setTeamId(Long teamId) { this.teamId = teamId; }
    public String getObjectKey() { return objectKey; }
    public void setObjectKey(String objectKey) { this.objectKey = objectKey; }
    public String getFileType() { return fileType; }
    public void setFileType(String fileType) { this.fileType = fileType; }
    public String getTraceId() { return traceId; }
    public void setTraceId(String traceId) { this.traceId = traceId; }
    public OffsetDateTime getCreatedAt() { return createdAt; }
    public void setCreatedAt(OffsetDateTime createdAt) { this.createdAt = createdAt; }
}
